osu.Game/Overlays/DirectOverlay.cs

```diff
@@ -232,21 +251,49 @@ protected override void PopIn()
         private SearchBeatmapSetsRequest getSetsRequest;
 
         private readonly Bindable<string> currentQuery = new Bindable<string>(string.Empty);
+        private int currentPage = 1;
+        private bool isLastPageLoaded = false;
 
         private ScheduledDelegate queryChangedDebounce;
+        private ScheduledDelegate addPageDebounce;
         private PreviewTrackManager previewTrackManager;
 
         private void queueUpdateSearch(bool queryTextChanged = false)
         {
             BeatmapSets = null;
             ResultAmounts = null;
+            currentPage = 1;
+            isLastPageLoaded = false;
 
             getSetsRequest?.Cancel();
 
             queryChangedDebounce?.Cancel();
             queryChangedDebounce = Scheduler.AddDelayed(updateSearch, queryTextChanged ? 500 : 100);
         }
 
+        private void queueAddPage()
+        {
+            if (getSetsRequest != null)
+                return;
+
+            if (addPageDebounce != null)
+                return;
+            
+            if (isLastPageLoaded)
+                return;
```

Comment:
I don't honestly see much value to splitting the guards out like that, but that's subjective.